@here/harp-mapview/lib/MapViewAtmosphere.ts

```diff
@@ -0,0 +1,481 @@
+     */
+    set enabled(enable: boolean) {
+        // Check already disposed.
+        if (this.disposed) {
+            return;
+        }
+        if (this.m_enabled === enable) {
+            return;
+        }
+        this.m_enabled = enable;
+        const isAdded = MapViewAtmosphere.isPresent(this.m_mapView);
+        if (enable && !isAdded) {
+            this.addToScene(this.m_mapView.scene);
+            this.addToMapView(this.m_mapView);
+        } else if (!enable && isAdded) {
+            this.removeFromScene(this.m_mapView.scene);
+            this.removeFromMapView(this.m_mapView);
+        }
+    }
+
+    /**
+     * Returns the current atmosphere status, enabled or disabled.
+     */
+    get enabled(): boolean {
+        return this.m_enabled;
+    }
+
+    set lightMode(lightMode: AtmosphereLightMode) {
+        if (this.m_materialVariant !== AtmosphereShadingVariant.ScatteringShader) {
+            return;
+        }
+        const dynamicLight = lightMode === AtmosphereLightMode.LightDynamic;
+        if (this.m_groundMaterial !== undefined) {
+            const groundMat = this.m_groundMaterial as GroundAtmosphereMaterial;
+            groundMat.setDynamicLighting(dynamicLight);
+        }
+        if (this.m_skyMaterial !== undefined) {
+            const skyMat = this.m_skyMaterial as SkyAtmosphereMaterial;
+            skyMat.setDynamicLighting(dynamicLight);
+        }
+    }
+
+    /**
+     * Disposes allocated resources.
+     */
+    dispose() {
+        // Unlink from scene and mapview anchors
+        if (this.enabled) {
+            this.enabled = false;
+        }
+
+        this.m_skyMaterial?.dispose();
+        this.m_groundMaterial?.dispose();
+
+        this.m_skyGeometry?.dispose();
+        this.m_groundGeometry?.dispose();
+
+        // After disposal we may no longer enable effect.
+        this.m_skyGeometry = undefined;
+        this.m_groundGeometry = undefined;
+
+        this.m_skyMaterial = undefined;
+        this.m_groundMaterial = undefined;
+
+        this.m_skyMesh = undefined;
+        this.m_groundMesh = undefined;
+    }
+
+    /**
+     * Sets the atmosphere depending on the [[Theme]] instance provided.
+     *
+     * This function is called when a theme is loaded. Atmosphere is added only if the theme
+     * contains a atmosphere definition with a:
+     * - `color` property, used to set the atmosphere color.
+     *
+     * @param theme A [[Theme]] instance.
+     */
+    reset(theme: Theme) {
+        //this.m_cachedTheme = theme;
+    }
+
+    /**
+     * Updates the atmosphere at runtime, depending on the camera and projection settings.
+     *
+     * @param camera An instance of a `THREE.Camera`.
+     * @param projection An instance of Projection currently in use.
+     */
+    update(camera: THREE.PerspectiveCamera | THREE.OrthographicCamera, projection: Projection) {
+        assert(
+            projection.type === this.m_mapView.projection.type,
```

Comment:
I guess this is called from MapView, so the check will always return true (i.e. projection and this.m_mapView.projection are the same).
One more reason not to keep map view

Reply:
This method will be gonne. It is not required and neither used, I though to find some way for MapViewAtmosphere (global) update, but this would probably require creating reversed dependency MapView -> MapViewAtmosphere, which I am trying to avoid.

Reply:
I think we could _integrate_ all so called MapView effects (or scene effects) in the same way, as you proposed in previous comments. Although I am not sure if it should be a part of this -> already big PR.

Reply:
I'm fine with merging this one and create a follow up ticket to align MapViewFog and MapViewAtmosphere